ecl/eclagent/eclagent.cpp

```diff
@@ -565,7 +566,23 @@ EclAgent::EclAgent(IConstWorkUnit *wu, const char *_wuid, bool _checkVersion, bo
         updateSuppliedXmlParams(w);
     }
     IPropertyTree *costs = queryCostsConfiguration();
-    agentMachineCost = costs ? costs->getPropReal("@agent", 0.0): 0.0;
+    if (costs)
+    {
+        agentMachineCost = money2cost_type(costs->getPropReal("@agent"));
+        if (agentMachineCost)
+        {
+            double softCostLimit = costs->getPropReal("@limit");
+            double guillotineCost = wu->getDebugValueReal("maxCost", softCostLimit);
+            if (guillotineCost > 0.0)
```

Comment:
This means if a user sets maxCost to 0 in a workunit they can avoid the hard cost limit.